core/src/main/java/org/ehcache/core/EhcacheManager.java

```diff
@@ -523,24 +537,8 @@ public void deregisterListener(CacheManagerListener listener) {
   @Override
   public void init() {
     final StatusTransitioner.Transition st = statusTransitioner.init();
-
-    if (serviceLocator.getService(CacheManagerProviderService.class) == null) {
-      this.serviceLocator.addService(new DefaultCacheManagerProviderService(this));
-    }
-
     try {
-      for (ServiceCreationConfiguration<? extends Service> serviceConfig : configuration.getServiceCreationConfigurations()) {
-        Service service = serviceLocator.getOrCreateServiceFor(serviceConfig);
-        if (service == null) {
-          throw new IllegalArgumentException("Couldn't resolve Service " + serviceConfig.getServiceType().getName());
-        }
-      }
-      serviceLocator.loadDependenciesOf(ServiceDeps.class);
-      try {
-        serviceLocator.startAllServices();
-      } catch (Exception e) {
-        throw st.failed(e);
-      }
+      serviceLocator.startAllServices();
```

Comment:
Removing the surrounding catch block will cause an exception to be triggered when trying to stop in the lower catch block which will end up shadowing the original exception.
So AFAICT the try/catch block is required here.